### PR TITLE
feat(svm): Publish SVM codes to Community

### PR DESCRIPTION
**Pull request by smrutis1**
Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

SVM code is an wrapper written over the sw360 community CVE service, SVM code enables sw360 server to connect with the SVM server, over the exposed SVM endpoints. Sw360 also uses SVM services to sync with the SVM server. Contributing the SVM specific code to community will benefit all the organization/s that uses SVM server.

> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 

This feature is configurable via the flag `enable.security.vulnerability.monitoring` in sw360.properties. 
Put the SVM related configs in sw360.properties file like (`svm.components.url`,  `svm.components.vulnerabilities.url` etc.) and check if the SVM functionalities working fine or not.

> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/admin/ScheduleAdminPortlet.java [hunk 1]
```diff
@@ -109,28 +124,112 @@ private boolean isAnyServiceScheduled(ScheduleService.Iface scheduleClient, User
...
+
+    @UsedAsLiferayAction
+    public void triggerSvmSync(ActionRequest request, ActionResponse response) throws PortletException, IOException {
         try {
-            User user = UserCacheHolder.getUserFromRequest(request);
-            RequestSummary requestSummary =
-                    new ThriftClients().makeScheduleClient().scheduleService(ThriftClients.CVESEARCH_SERVICE);
-            setSessionMessage(request, requestSummary.getRequestStatus(), "Task", "schedule");
+            RequestStatus requestStatus = new ThriftClients().makeVMClient().synchronizeComponents().getRequestStatus();
+            setSessionMessage(request, requestStatus, "Task", "perform");
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/admin/ScheduleAdminPortlet.java:149]: change `perform` to `performe`.
also at line `169`, `193` and `222`.

File: libraries/datahandler/src/main/thrift/vmcomponents.thrift [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+service VMComponentService {
+    // General information
+    list<VMProcessReporting> getAllProcesses(1: User user);
+    list<VMMatch> getAllMatches(1: User user);
+
+    // Trigger for vulnerability monitoring
+    RequestSummary synchronizeComponents(/*1: User user*/);
+    RequestSummary triggerReverseMatch(/*1: User user*/);
+//    RequestSummary scheduleSync(1: User user);
+//    RequestSummary unscheduleSync(1: User user);
```
Comment (akapti) [inline libraries/datahandler/src/main/thrift/vmcomponents.thrift:147]: remove the commented lines (`146` and `147`). also the commented method parameters in line `144` and `145`.

File: frontend/sw360-portlet/src/main/webapp/html/admin/vulnerabilityManagement/reporting.jspf [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        <tfoot>
+        <tr>
+            <th colspan="4"></th>
+        </tr>
+        </tfoot>
+    </table>
+</div>
+
+<script type="text/javascript">
+    require(['jquery', /* jquery-plugins */ 'datatables.net'], function ($) {
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/webapp/html/admin/vulnerabilityManagement/reporting.jspf:29]: remove the commented code `jquery-plugins`

File: frontend/sw360-portlet/src/main/webapp/html/admin/vulnerabilityManagement/matches.jspf [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        <tfoot>
+        <tr>
+            <th colspan="5"></th>
+        </tr>
+        </tfoot>
+    </table>
+</div>
+
+<script type="text/javascript">
+    require(['jquery', /* jquery-plugins */ 'datatables.net'], function ($) {
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/webapp/html/admin/vulnerabilityManagement/matches.jspf:34]: remove the commented code `jquery-plugins`

File: backend/src/src-vmcomponents/src/main/java/org/eclipse/sw360/vmcomponents/handler/SVMSyncHandler.java [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+import org.eclipse.sw360.datahandler.thrift.vulnerabilities.Vulnerability;
+import org.eclipse.sw360.vmcomponents.common.SVMConstants;
+import org.eclipse.sw360.vmcomponents.common.SVMMapper;
+import org.eclipse.sw360.vmcomponents.common.SVMUtils;
+import org.eclipse.sw360.vmcomponents.common.VMResult;
+import org.eclipse.sw360.vmcomponents.db.VMDatabaseHandler;
+import org.eclipse.sw360.vmcomponents.process.VMProcessHandler;
+import org.eclipse.sw360.vulnerabilities.common.VulnerabilityMapper;
+import org.jetbrains.annotations.NotNull;
+import org.json.simple.JSONArray;
```
[the remark this answers is not in the document]
Comment (akapti) [reply backend/src/src-vmcomponents/src/main/java/org/eclipse/sw360/vmcomponents/handler/SVMSyncHandler.java:32]: from discussion: this require more changes, and will be taken care in future.

File: backend/src/src-vmcomponents/src/main/java/org/eclipse/sw360/vmcomponents/common/SVMMapper.java [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+import org.apache.log4j.Logger;
+import org.eclipse.sw360.datahandler.common.SW360Constants;
+import org.eclipse.sw360.datahandler.common.SW360Utils;
+import org.eclipse.sw360.datahandler.thrift.components.Component;
+import org.eclipse.sw360.datahandler.thrift.components.Release;
+import org.eclipse.sw360.datahandler.thrift.vulnerabilities.CVEReference;
+import org.eclipse.sw360.datahandler.thrift.vulnerabilities.VendorAdvisory;
+import org.eclipse.sw360.datahandler.thrift.vulnerabilities.Vulnerability;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
```
[the remark this answers is not in the document]
Comment (akapti) [reply backend/src/src-vmcomponents/src/main/java/org/eclipse/sw360/vmcomponents/common/SVMMapper.java:19]: from discussion: this require more changes, and will be taken care in future.

File: backend/src-common/src/main/resources/sw360.properties [hunk 7]
```diff
@@ -70,6 +70,41 @@ textForNewRelease= a new release %s %s, in which you take part, has been created
...
+# default is 02:00 am (2*60*60=7200)
+schedule.svmmatch.firstOffset.seconds=7200
+# default is 24h (24*60*60sec=86400)
+schedule.svmmatch.interval.seconds=86400
+
+# default is 03:00 am (3*60*60=10800)
+schedule.svmlistupdate.firstOffset.seconds=10800
+# default is 24h (24*60*60sec=86400)
+schedule.svmlistupdate.interval.seconds=86400
+
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply backend/src-common/src/main/resources/sw360.properties:96]: Done

**Review by akapti — APPROVED** [review]
LGTM.

**Review by ag4ums — APPROVED** [review]
code looks good working as expected.

**Pull request merged**